.goreleaser.yaml

```diff
@@ -0,0 +1,33 @@
+# yaml-language-server: $schema=https://goreleaser.com/static/schema.json
+project_name: imagebuilder
+before:
+  hooks:
+    - go mod tidy
+    - go generate ./...
+builds:
```

Comment:
I think you need to define the file containing `main()`, e.g. like [this](https://github.com/GoTestTools/gotestfmt/blob/9eae5abc81d6d08f73268741ef4a8b97f29b60d8/.goreleaser.yml#L12). So in this case it would be
```yaml
builds:
   ...
   main: ./carpenter.go   
```